Add missing specs for sql manager page (#191)
PrestaShop/prestashop-specs: back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md

"When saving the query, only the "SELECT" SQL statement is allowed."

### SQL Manager

#### Settings wheel

When clicking on it, three options are displayed:

1) Refresh list

> After clicking on it, the page is reloaded. Sorts and searches are keeped.

Reviewer: ```suggestion
After clicking on it, the page is reloaded. Sorts and searches are kept.
```

Author: My bad :scream:

2) Show SQL query

After clicking on it, the SQL query is displayed in a modal. The SQL query takes into account filters and sorts that have been done.

There are two buttons:

- Close: After clicking on it, the modal is closed
- Export to SQL Manager: After clicking on it, the SQL query is opened in Advanced parameters > Database > SQL Manager

3) Export to SQL Manager

After clicking on it, the SQL query is opened in Advanced parameters > Database > SQL Manager

#### Bulk actions

> Clicking on the _Bulk actions_ button will give the user the ability to perform bulk actions for selected sql queries (using the check boxes). The user can:

Reviewer: ```suggestion
Clicking on the _Bulk actions_ button allows users to perform bulk actions for selected SQL queries (using the checkboxes), or query - the button is clickable starting from one selected item. A user can:
```

> - Select all/Unselect all: This button will select/unselect all check boxes next to each row on the table.

Reviewer: ```suggestion

- Select all/Unselect all: this button will select/unselect all the checkboxes beginning each row of the listing.
```

- Delete selected: This button will delete the selected sql queries. 
After clicking on "Delete selected", a modal with title "Delete selection" is opened asking to confirm or cancel the action: "Are you sure you want to delete the selected item(s)?" (See issue [#14462](https://github.com/PrestaShop/PrestaShop/issues/14462))

The action can be canceled by clicking on the cross or on the cancel button.

After clicking on "Delete" button, the selected webservice keys are deleted, a successful message is displayed "The selection has been successfully deleted."

#### Columns

The page displays the SQL Manager list with the following columns:

1) ID

It's possible to sort by id asc or desc and search by id.

2) SQL query name

It's possible to sort by SQL query name asc or desc and search by SQL query name.

3) SQL query 

It's possible to sort by SQL query asc or desc and search by SQL query.

4) Actions 

- View. Executes the query, and get the results displayed in an HTML table, right within the PrestaShop interface. This is the main action and it can be performed by clicking on the line.

> When the user views the SQL request with no result, a warning is displayed: "This SQL query has no result."

Reviewer: Should it be considered as an error or something that is not right? I'm wondering if this shouldn't be turned into a simple information notification. 🤔

Author: Humm, it's not an error, it's just that there is no results
Like when you search something in a list and there are no results

Reviewer: Yep, then I'd be in favor of turning it into an information notification. It looks less scary.

> - Export. Executes the query, and downloaded the result in CSV format.

Reviewer: Since this action is no longer part of the list but an icon before the actions list, I'd notice it before the View / Edit / Delete panel.

Author: You're right :)

- Edit. The user can edit a query as often as necessary, to refine it and get better results.
- Delete. Once a query is not used anymore (or simply because it does not work), you can delete it by clicking this button and confirming your choice.
After clicking on _Delete_, a modal with title "Delete selection" is opened asking to confirm or cancel the action: "Are you sure you want to delete the selected item(s)?" 

The action can be canceled by clicking on the cross or on the cancel button.

After clicking on "Delete" button, the message is deleted and a successful message is displayed "Successful deletion."

#### Multistore behavior

> As SQL query can't be associated to specific shop(s) during the creation, so all the SQL query from all stores are listed, no matter the context selected.

Reviewer: I'm not sure I understand: a SQL query cannot be dedicated to a specific shop? By default, it affects all shops?

Author: Yes, it affects all shops, you can't create an SQK query just for one shop

Reviewer: ```suggestion
By default, a SQL query affects all shops. Since it can't be associated with any specific shop, all the SQL queries listed on this page are used in an all shops context.
```

### Settings

Select your default file encoding. You can configure the character encoding of the downloaded CSV file. The default, UTF-8, is recommended, but you can select ISO-8859-1 if need be through the drop-down.

#### Multistore behavior

In a specific shop context or in a group context, the "default file encoding" doesn't have checkbox and is disabled with an information message displayed when the user hovers the drop-down: "You can modify this parameter in all shops context only. Switch context to enable it." (see issue [#19356](https://github.com/PrestaShop/PrestaShop/issues/19356))

> In an all shops context, the the "default file encoding" doesn't have specific settings drop-down.

Reviewer: ```suggestion
In an all shops context, this feature is disabled. Merchants cannot select any file encoding.
```

Author: Humm, that's not the case, precisely, it can be changed only in all shop context.
The specific settings drop-down is one of the new components that will be integrated in 1.7.8 and allowing in all shops context to see for a parameter if it has a different value in a specific shop.
But as the file encoding has the same value for all shops, it will not be displayed

Reviewer: Noted, thanks, I might have read the multistore specifications too quickly. ;-)

Each keywords are checked in the SQL query before saving it.

On the bottom left, the "List of MySQL Tables" gives the user a list of all the available database tables. He/she should select a table to make PrestaShop display all its attributes and types in the list on the right, then click "Add attribute to SQL query" or "Add table name to SQL query" to send its attribute name/table name into the SQL query field.

#### Multistore behavior

> This page has no shop association block. In this case, the sql query is added for all shops, no matter the context selected.

Reviewer: ```suggestion
This page has no shop association block. By default, the SQL query is added for all shops, no matter the context selected.
```
